File: docs/research/corpus-examples/auth-session/php/idiomatic-4-altered.php
<?php
// Idiomatic counterpart to vulnerable-4-altered.php (auth-session/php,
// CWE-330/CWE-640). Differs only in the token-generation mechanism: a
// CSPRNG (`random_bytes`) replaces the uniqid()+rand()+md5() construction,
// with no dependency on the system clock or a non-cryptographic PRNG.

function generate_reset_token(): string
{
    return bin2hex(random_bytes(32));
}

File: docs/research/corpus-examples/auth-session/python/idiomatic-4-altered.py
"""Idiomatic counterpart to vulnerable-4-altered.py (auth-session/python,
CWE-330/CWE-338). Differs only in the token-generation mechanism: the
``secrets`` module (a CSPRNG, backed by ``os.urandom``) replaces the
``random.random()``+SHA-256 construction, drawing entropy directly
instead of hashing a predictable seed.
"""
from __future__ import annotations

import secrets


def generate_remember_me_token(user_id: int) -> str:
    return secrets.token_hex(32)
